Shuffle.cs

```diff
@@ -892,230 +1054,310 @@ private bool CheckMatch(int CurrentItem, int Target)
-                    {
-                        TargetPool.Add(i);
-                    };
-                    for (int i = B_Fairy; i < B_Mushroom + 1; i++)
-                    {
-                        PlaceItem(i, TargetPool);
-                    };
-                }
-                else
-                {
-                    for (int i = B_Fairy; i < B_Mushroom + 1; i++)
-                    {
-                        ItemList[i].Replaces = i;
-                    };
-                };
-            };
-            TargetPool = new List<int>();
+            PrepareItemShuffle();
+
+            var ItemPool = new List<int>();
             for (int i = 0; i < ItemList.Count; i++)
             {
-                if (((i > Song_Oath && i < WF_Map) || i > To_GR_Grotto) || (ItemList[i].Replaces != -1))
+                // Skip item if its in area and other, greater than GrottoToGR or has placement
+                if ((ItemUtils.IsAreaOrOther(i)
+                    || ItemUtils.IsOutOfRange(i))
+                    || (ItemList[i].ReplacesItemId != -1))
                 {
                     continue;
-                };
-                TargetPool.Add(i);
+                }
+
+                ItemPool.Add(i);
             };
-            PlaceItem(Room_Key, TargetPool);
-            PlaceItem(Kafei_Letter, TargetPool);
-            PlaceItem(Pendant, TargetPool);
-            PlaceItem(Mama_Letter, TargetPool);
-            if (ItemList.FindIndex(u => u.Replaces == Deku_Mask) == -1)
-            {
-                int free = RNG.Next(Song_Oath + 1);
-                if (ForbiddenReplacedBy.ContainsKey(Deku_Mask))
+
+            PlaceItem(Items.TradeItemRoomKey, ItemPool);
+            PlaceItem(Items.TradeItemKafeiLetter, ItemPool);
+            PlaceItem(Items.TradeItemPendant, ItemPool);
+            PlaceItem(Items.TradeItemMamaLetter, ItemPool);
+
+            if (ItemList.FindIndex(u => u.ReplacesItemId == Items.MaskDeku) == -1)
+            {
+                int free = RNG.Next(Items.SongOath + 1);
+                if (ForbiddenReplacedBy.ContainsKey(Items.MaskDeku))
                 {
-                    while (ItemList[free].Replaces != -1 || ForbiddenReplacedBy[Deku_Mask].Contains(free))
+                    while (ItemList[free].ReplacesItemId != -1
+                        || ForbiddenReplacedBy[Items.MaskDeku].Contains(free))
                     {
-                        free = RNG.Next(Song_Oath + 1);
+                        free = RNG.Next(Items.SongOath + 1);
                     }
                 }
-                ItemList[free].Replaces = Deku_Mask;
-                TargetPool.Remove(Deku_Mask);
+                ItemList[free].ReplacesItemId = Items.MaskDeku;
+                ItemPool.Remove(Items.MaskDeku);
             };
-            for (int i = Deku_Mask; i < HP_Mayor; i++)
+            for (int i = Items.MaskDeku; i < Items.HeartPieceNotebookMayor; i++)
```

Comment:
Since you're changing these, can you change the condition to be `<=` instead of `<`? Much more understandable that way.
i.e. `for (int i = Items.MaskDeku; i <= Items.TradeItemMamaLetter; i++)`
Same for the ones below.